src/cascadia/TerminalApp/ColorScheme.h

```diff
@@ -34,28 +34,27 @@ namespace winrt::TerminalApp::implementation
     public:
         ColorScheme();
         ColorScheme(hstring name, Windows::UI::Color defaultFg, Windows::UI::Color defaultBg, Windows::UI::Color cursorColor);
-        ~ColorScheme();
 
         static com_ptr<ColorScheme> FromJson(const Json::Value& json);
         bool ShouldBeLayered(const Json::Value& json) const;
         void LayerJson(const Json::Value& json);
 
-        hstring Name() const noexcept;
-        com_array<Windows::UI::Color> Table() const noexcept;
-        Windows::UI::Color Foreground() const noexcept;
-        Windows::UI::Color Background() const noexcept;
-        Windows::UI::Color SelectionBackground() const noexcept;
-        Windows::UI::Color CursorColor() const noexcept;
+        static Json::Value ToJson(const TerminalApp::ColorScheme& scheme);
+        void UpdateJson(Json::Value& json);
 
         static std::optional<std::wstring> GetNameFromJson(const Json::Value& json);
 
+        com_array<Windows::UI::Color> Table() const noexcept;
+        void SetColorTableEntry(uint8_t index, const winrt::Windows::UI::Color& value) noexcept;
+
+        GETSET_PROPERTY(winrt::hstring, Name, L"");

```

Comment:
can we get away with zero-initializing this and not emitting an empty string constant?

Michael has mentioned it a few times, but maybe we should establish a rule:

_if you find yourself using `""` or `L""`, don't._